presto-hive/src/main/java/com/facebook/presto/hive/HivePartition.java

```diff
@@ -117,4 +146,71 @@ public String toString()
     {
         return tableName + ":" + partitionId;
     }
+
+    /**
+     * This helper class makes it easy to serialize/deserialize map objects in JSON.
+     */
+    public static class JsonSerializableMapEntries<K, V>
+    {
+        private final List<Entry<K, V>> list;
+
+        @JsonCreator
+        public JsonSerializableMapEntries(
+                @JsonProperty("list") List<Entry<K, V>> list)
+        {
+            this.list = list;
+        }
+
+        @JsonProperty
+        public List<Entry<K, V>> getList()
+        {
+            return list;
+        }
+
+        public static <K, V> JsonSerializableMapEntries<K, V> toEntries(Map<K, V> map)
+        {
+            ImmutableList.Builder<Entry<K, V>> builder = ImmutableList.builder();
+
+            for (Map.Entry<K, V> entry : map.entrySet()) {
+                builder.add(new Entry<>(entry.getKey(), entry.getValue()));
+            }
+
+            return new JsonSerializableMapEntries<>(builder.build());
+        }
+
+        public static <K, V> Map<K, V> toMap(JsonSerializableMapEntries<K, V> entries)
+        {
+            ImmutableMap.Builder<K, V> builder = ImmutableMap.builder();
+            for (Entry<K, V> entry : entries.list) {
+                builder.put(entry.getKey(), entry.getValue());
+            }
+            return builder.build();
+        }
+
+        public static class Entry<K, V>
```

Comment:
It sounds like Jackson 2.5 might have added serialization/deserialization from `Map.Entry`. Can you try that out? https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.5